### ISSUE TITLE
Track order bug when adding a srt stream to an existing MP4 with MP4BOX

### ISSUE THREAD

**Issue by LeeThompson** [gpac/gpac#1777]
Software Version: `MP4Box - GPAC version 1.0.1-rev0-gd8538e8a-master`
OS Version: `Microsoft Windows [Version 10.0.19041.928]`

This is a bit of an odd one, I'm adding captions (SRT format) to existing MP4 files, the ones _not_ initially created by MP4BOX work fine.    However, the 10 files that I processed with MP4BOX (was adding commentary tracks) previously, strange things happened to the track order.

My command line for building them initially was:
`mp4box -new "filename.mp4" -add "assets\filename.hvc:name=" -add "assets\filename.aac:name=:lang=eng" -add "commentary\filename.aac:name=Commentary with Cast and Crew:lang=eng:delay=0"`

This worked fine, but now adding captions with:
`mp4box "filename.mp4" -add "filename.srt:name=:lang=eng"`

It adds the captions but now the track order is putting the commentary first.  The track number remains 3 but it appears second and most software defaults to it.   Very strange.     Weirder, the files that were encoded and muxed by Handbrake do not exhibit this behavior.  (Although they only had the video and single audio track.)

**Initial File**
```
Format                                   : MPEG-4
Format profile                           : Base Media
Codec ID                                 : iso4 (iso4/iso6)
File size                                : 292 MiB
Duration                                 : 26 min 55 s
Overall bit rate mode                    : Variable
Overall bit rate                         : 1 518 kb/s
Encoded date                             : UTC 2021-05-01 04:31:12
Tagged date                              : UTC 2021-05-01 04:31:12

Video
ID                                       : 1
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main@L3.1@Main
Codec ID                                 : hvc1
Codec ID/Info                            : High Efficiency Video Coding
Duration                                 : 26 min 55 s
Bit rate                                 : 1 285 kb/s
Maximum bit rate                         : 7 447 kb/s
Width                                    : 1 280 pixels
Height                                   : 720 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Bits/(Pixel*Frame)                       : 0.058
Stream size                              : 247 MiB (85%)
Encoded date                             : UTC 2021-05-01 04:31:12
Tagged date                              : UTC 2021-05-01 04:31:19
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
Codec configuration box                  : hvcC

Audio #1
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 26 min 55 s
Bit rate mode                            : Variable
Bit rate                                 : 128 kb/s
Maximum bit rate                         : 135 kb/s
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 44.1 kHz
Frame rate                               : 43.066 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 24.6 MiB (8%)
Language                                 : English
Encoded date                             : UTC 2021-05-01 04:31:16
Tagged date                              : UTC 2021-05-01 04:31:19

Audio #2
ID                                       : 3
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 26 min 55 s
Bit rate mode                            : Variable
Bit rate                                 : 100.0 kb/s
Maximum bit rate                         : 106 kb/s
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 44.1 kHz
Frame rate                               : 43.066 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 19.3 MiB (7%)
Title                                    : Commentary with Cast and Crew
Language                                 : English
Encoded date                             : UTC 2021-05-01 04:31:16
Tagged date                              : UTC 2021-05-01 04:31:19
```

**Updated File**
```
Format                                   : MPEG-4
Format profile                           : Base Media
Codec ID                                 : iso4 (iso4/iso6)
File size                                : 292 MiB
Duration                                 : 26 min 55 s
Overall bit rate mode                    : Variable
Overall bit rate                         : 1 518 kb/s
Encoded date                             : UTC 2021-05-01 04:31:12
Tagged date                              : UTC 2021-05-01 04:31:12

Video
ID                                       : 1
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main@L3.1@Main
Codec ID                                 : hvc1
Codec ID/Info                            : High Efficiency Video Coding
Duration                                 : 26 min 55 s
Bit rate                                 : 1 285 kb/s
Maximum bit rate                         : 7 447 kb/s
Width                                    : 1 280 pixels
Height                                   : 720 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Bits/(Pixel*Frame)                       : 0.058
Stream size                              : 247 MiB (85%)
Encoded date                             : UTC 2021-05-01 04:31:12
Tagged date                              : UTC 2021-05-05 17:34:50
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
Codec configuration box                  : hvcC

Audio #1
ID                                       : 3
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 26 min 55 s
Bit rate mode                            : Variable
Bit rate                                 : 100.0 kb/s
Maximum bit rate                         : 106 kb/s
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 44.1 kHz
Frame rate                               : 43.066 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 19.3 MiB (7%)
Title                                    : Commentary with Cast and Crew
Language                                 : English
Encoded date                             : UTC 2021-05-01 04:31:18
Tagged date                              : UTC 2021-05-05 17:34:50

Audio #2
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 26 min 55 s
Bit rate mode                            : Variable
Bit rate                                 : 128 kb/s
Maximum bit rate                         : 135 kb/s
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 44.1 kHz
Frame rate                               : 43.066 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 24.6 MiB (8%)
Language                                 : English
Encoded date                             : UTC 2021-05-01 04:31:16
Tagged date                              : UTC 2021-05-05 17:34:50

Text
ID                                       : 4
Format                                   : Timed Text
Codec ID                                 : tx3g
Duration                                 : 26 min 48 s
Bit rate mode                            : Variable
Bit rate                                 : 120 b/s
Stream size                              : 23.5 KiB (0%)
Language                                 : English
Forced                                   : No
Encoded date                             : UTC 2021-05-05 17:34:50
Tagged date                              : UTC 2021-05-05 17:34:50
```

**Comment by jeanlf** [comment]
I believe this was already fixed (cf #1686), could you check with latest git ?

**Comment by LeeThompson** [comment]
Confirmed fixed.  
Tested `MP4Box - GPAC version 1.1.0-DEV-rev802-gc0ea96c7-master`
